lib/ssl/src/ssl_connection.erl

```diff
@@ -1179,6 +1179,34 @@ certify(internal, #client_key_exchange{exchange_keys = Keys},
 	#alert{} = Alert ->
 	    handle_own_alert(Alert, Version, ?FUNCTION_NAME, State)
     end;
+%% The response will be handled only when a certificate status request has
+%% been sent by the client and confirmed by the server.
+certify(internal, #certificate_status{response = OcspRespDer},
+        #state{handshake_env = #handshake_env{
+                   ocsp_stapling_state = #{
+                       is_client_hello_requested := true,
```

Comment:
See my comment about this variable in tls_connection.erl. I think you can use the ssl option here that enables stapling. A compliant ssl server will only send a `certificate_status` if the client has sent a CertificateStatusRequest extension.
I also think we don't need the is_server_hello_requested variable in this case, and this way the ocsp_stapling_state can be expressed by using the above mentioned ssl option that enables ocsp stapling and the cert_db/ocsp_cert_db in static_env.

Reply:
> See my comment about this variable in tls_connection.erl. I think you can use the ssl option here that enables stapling. A compliant ssl server will only send a `certificate_status` if the client has sent a CertificateStatusRequest extension.
> I also think we don't need the is_server_hello_requested variable in this case, and this way the ocsp_stapling_state can be expressed by using the above mentioned ssl option that enables ocsp stapling and the cert_db/ocsp_cert_db in static_env.

how we ensure the following in rfc6066 in this case?

If a server returns a "CertificateStatus" message, then the server MUST have included an extension of type 
"status_request" with empty "extension_data" in the extended server hello.